codes/ruby/chapter_hashing/hash_map_open_addressing.rb

```diff
@@ -0,0 +1,146 @@
+=begin
+File: hash_map_open_addressing.rb
+Created Time: 2024-04-13
+Author: Xuan Khoa Tu Nguyen (ngxktuzkai2000@gmail.com)
+=end
+
+require_relative './array_hash_map'
+
+### 开放寻址哈希表 ###
+class HashMapOpenAddressing
+  TOMBSTONE = Pair.new(-1, '-1') # 删除标记
+
+  ### 构造方法 ###
+  def initialize
+    @size = 0 # 键值对数量
+    @capacity = 4 # 哈希表容量
+    @load_thres = 2.0 / 3.0 # 触发扩容的负载因子阈值
+    @extend_ratio = 2 # 扩容倍数
+    @buckets = Array.new(@capacity) # 桶数组
+  end
+
+  ### 哈希函数 ###
+  def hash_func(key)
+    key % @capacity
+  end
+
+  ### 负载因子 ###
+  def load_factor
+    @size / @capacity
+  end
+
+  ### 搜索 key 对应的桶索引 ###
+  def find_bucket(key)
+    index = hash_func(key)
+    first_tombstone = -1
+    # 线性探测，当遇到空桶时跳出
+    while !@buckets[index].nil?
+      # 若遇到 key ，返回对应的桶索引
+      if @buckets[index].key == key
+        # 若之前遇到了删除标记，则将建值对移动至该索引处
```

Comment:
There is a typo here.

```suggestion
        # 若之前遇到了删除标记，则将键值对移动至该索引处
```